storm-server/src/main/java/org/apache/storm/healthcheck/HealthChecker.java

```diff
@@ -66,10 +71,23 @@ public static int healthCheck(Map<String, Object> conf) {
         // to execute properly, not that the system is unhealthy, in which case
         // we don't want to start killing things.
 
-        if (results.contains(FAILED) || results.contains(FAILED_WITH_EXIT_CODE)
-            || results.contains(TIMEOUT)) {
+        if (results.contains(FAILED) || results.contains(FAILED_WITH_EXIT_CODE)) {
             LOG.warn("The supervisor healthchecks failed!!!");
             return 1;
+        } else if (results.contains(TIMEOUT)) {
+            LOG.warn("The supervisor healthchecks timedout!!!");
+            if (supervisor != null) {
+                Meter timeoutMeter = supervisor.getMetricsRegistry().getMeter(Constants.SUPERVISOR_HEALTH_CHECK_TIMEOUTS);
+                if (timeoutMeter != null) {
+                    timeoutMeter.mark();
+                }
+            }
+            Boolean failOnTimeouts = ObjectReader.getBoolean(conf.get(DaemonConfig.STORM_HEALTH_CHECK_FAIL_ON_TIMEOUTS), true);
```

Comment:
We can remove this since it's already in the `default.yaml` file to avoid duplicates.
 
Or remove the one in default.yaml